include/itkAnalyticSignalImageFilter.h

```diff
@@ -85,10 +88,24 @@ class AnalyticSignalImageFilter:
       {
       this->m_FFTRealToComplexFilter->SetDirection( direction );
       this->m_FFTComplexToComplexFilter->SetDirection( direction );
+      if( this->m_FrequencyFilter.IsNotNull() )
+        {
+        this->m_FrequencyFilter->SetDirection( direction );
+        } 
       this->Modified();
       }
     }
 
+  virtual void SetFrequencyFilter(typename FrequencyFilterType::Pointer filter)
```

Comment:
It is best to use raw pointers in the interfaces to avoid creation/deletion of `SmartPointers` and for wrapping reasons, so

```
virtual void SetFrequencyFilter( FrequencyFilterType * filter)
```